shared/src/main/scala/async/Async.scala

```diff
@@ -30,7 +30,7 @@ import scala.util.boundary
   * @see
   *   [[Async$.group Async.group]] and [[Future$.apply Future.apply]] for [[Async]]-subscoping operations.
   */
-trait Async(using val support: AsyncSupport, val scheduler: support.Scheduler):
+trait Async private[async] (using val support: AsyncSupport, val scheduler: support.Scheduler):
```

Comment:
Should we make it sealed?

Reply:
Provide an adapter for extension point ？

Reply:
I am thinking about whether `Async` should be extensible or not. By not doing it, we retain more power to reason about it. It will also make it easier to add more to it without breaking changes (for example, context-local data)